pepper-apis/src/main/java/org/broadinstitute/ddp/housekeeping/handler/EmailNotificationHandler.java

```diff
@@ -149,6 +162,31 @@ public void handleMessage(NotificationMessage message) {
         }
     }
 
+    private Map<String, Object> getDynamicData(NotificationMessage message) {
+        Map<String, Object> dynamicData = new HashMap<>();
+
+        String salutation = generateSalutation(
+                message.getParticipantFirstName(),
+                message.getParticipantLastName(),
+                message.getDefaultSalutation());
+
+        dynamicData.put(SALUTATION, salutation);
+        dynamicData.put(BASE_WEB_URL, message.getWebBaseUrl());
+        dynamicData.put(STUDY_GUID, message.getStudyGuid());
+        dynamicData.put(PARTICIPANT_GUID, message.getParticipantGuid());
+        dynamicData.put(PARTICIPANT_FIRST_NAME, message.getParticipantFirstName());
+        dynamicData.put(PARTICIPANT_LAST_NAME, message.getParticipantLastName());
+
+        for (NotificationTemplateSubstitutionDto sub : message.getTemplateSubstitutions()) {
+            if (!sub.getVariableName().startsWith("-")) {
+                dynamicData.put(sub.getVariableName(), sub.getValue());
```

Comment:
I don't think this will do what we want? The queued event message will have template substitutions like `-ddp.link-` that needs to go in the dynamic template as well.